packages/@divvi/mobile/src/divviProtocol/slice.ts

```diff
@@ -0,0 +1,49 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { getRehydratePayload, REHYDRATE, RehydrateAction } from 'src/redux/persist-helper'
+import { RootState } from 'src/redux/store'
+
+interface ReferralState {
+  successfulReferrals: {
+    [key: string]: boolean // key is `${divviId}-${campaignIds.join(',')}` where campaignIds are sorted
+  }
+}
+
+const initialState: ReferralState = {
+  successfulReferrals: {},
+}
+
+const divviProtocolSlice = createSlice({
+  name: 'divviProtocol',
+  initialState,
+  reducers: {
+    markReferralSuccessful: (
+      state,
+      action: PayloadAction<{ divviId: string; campaignIds: string[] }>
+    ) => {
+      const { divviId, campaignIds } = action.payload
+      const sortedCampaignIds = [...campaignIds].sort()
+      const key = `${divviId}-${sortedCampaignIds.join(',')}`
+      state.successfulReferrals[key] = true
+    },
+  },
+  extraReducers: (builder) => {
+    builder.addCase(REHYDRATE, (state, action: RehydrateAction) => ({
+      ...state,
+      ...getRehydratePayload(action, 'divviProtocol'),
+    }))
+  },
```

Comment:
nit: we could omit this, the persisted state is restored automatically. It's only needed when we want to reset some specific props during rehydration, instead of keeping the persisted value.